packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2283,13 +2275,38 @@ export function commitLayoutEffects(
   committedLanes: Lanes,
 ): void {
   nextEffect = finishedWork;
-  commitLayoutEffects_begin(finishedWork, root, committedLanes);
+  commitLayoutEffects_intermediate(
+    finishedWork,
+    root,
+    committedLanes,
+    false,
+    false,
+  );
+}
+
+export function commitLayoutEffects_intermediate(
+  finishedWork: Fiber,
+  root: FiberRoot,
+  committedLanes: Lanes,
+  subtreeWasHidden: boolean,
+  subtreeIsHidden: boolean,
+): void {
+  nextEffect = finishedWork;
+  commitLayoutEffects_begin(
+    finishedWork,
+    root,
+    committedLanes,
+    subtreeWasHidden,
+    subtreeIsHidden,
```

Comment:
Instead of passing these as arguments, I would use module-level variables and mutate them before/after the `commitLayoutEffects_begin` in this function. Since this isn't the last of these types of contextual variables we'll want to use. Saves registers for the majority of frames where this isn't accessed.

Reply:
I would probably inline this function, too. There will be similar intermediate functions wherever we use this pattern.

Reply:
Sure that's fair.

Reply:
Oh I had already removed this. I didn't mean to commit it 😆